ledger/src/shred.rs

```diff
@@ -1129,9 +1129,8 @@ pub(crate) fn recover(
     // The same signature also verifies for recovered shreds because when
     // reconstructing the Merkle tree for the erasure batch, we will obtain the
     // same Merkle root.
-    merkle::recover(shreds, reed_solomon_cache)?
-        .map(|shred| Ok(Shred::from(shred?)))
-        .collect()
+    let shreds = merkle::recover(shreds, reed_solomon_cache)?;
```

Comment:
Outside of scope of this PR, but it looks like `merkle::recover()` needs a sorted list. And we need the shreds collected in order to know if they're sorted, so not sure we can avoid the the collect higher up in this function.